pulsar-broker/src/main/java/org/apache/pulsar/broker/TransactionMetadataStoreService.java

```diff
@@ -394,21 +402,32 @@ public void handleOpFail(Throwable e, TransactionCoordinatorID tcId) {
         }
     }
 
+    // after the completion of txn end, remove txnMeta.
+    public void removeTxnMeta(TxnID txnID) {
+        TransactionCoordinatorID tcId = getTcIdFromTxnId(txnID);
+        TransactionMetadataStore store = stores.get(tcId);
+        store.removeTxnMeta(txnID.getLeastSigBits());
+    }
+
     public void endTransactionForTimeout(TxnID txnID) {
         getTxnMeta(txnID).thenCompose(txnMeta -> {
             if (txnMeta.status() == TxnStatus.OPEN) {
-                return endTransaction(txnID, TxnAction.ABORT_VALUE, true);
+                return endTransaction(txnID, TxnAction.ABORT_VALUE, true)
+                        .thenRun(() -> removeTxnMeta(txnID));
             } else {
+                removeTxnMeta(txnID);
```

Comment:
the transaction status can be in committing status, why we need to remove TxnMeta?

Reply:
Not completed, i will ping you when finished.